File: 53-maximum-subarray/maximum-subarray.ts
//kadane's Algo

function maxSubArray(nums: number[]): number {
    let maxSum = -Infinity
    let sum = 0

    for (let i = 0; i < nums.length; i++) {
        sum += nums[i]

        maxSum = Math.max(sum, maxSum)

        if(sum < 0){
            sum = 0
        }
    }
    return maxSum
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = -float('inf')
        sum = 0

        for i in range(0, len(nums)):
            sum += nums[i]

            if sum > maxSum:
                maxSum = sum
            
            if sum < 0:
                sum = 0
        
        return maxSum
        
        